client/web/src/search/home/QueryExamplesHomepage.tsx

```diff
@@ -85,17 +99,83 @@ export const QueryExamplesHomepage: React.FunctionComponent<QueryExamplesHomepag
             setSelectedTip,
             selectTipTimeout,
             setSelectTipTimeout,
+            queryExampleTabActive,
+            history,
+            isSourcegraphDotCom
         ]
     )
 
     return (
         <div>
-            <div>
-                {isSourcegraphDotCom ? (
-                    <div className="d-flex align-items-center mb-2">
-                        <Text className={classNames('mr-2 pr-2', styles.codeBasicsTitle)}>Code Search Basics</Text>
+            {isSourcegraphDotCom ? (
+                <>
+                    <Tabs size="medium" onChange={handleTabChange}>
+                        <TabList wrapperClassName={classNames('mb-4', styles.tabHeader)}>
+                            <Tab key="Code search basics" className="pr-3">Code search basics</Tab>
+                            <Tab key="Search query examples">Search query examples</Tab>
+                        </TabList>
+                        <TabPanels>
+                            <TabPanel>
+                                <div className={styles.queryExamplesSectionsColumns}>
+                                    {exampleSyntaxColumns.map((column, index) => (
+                                        // eslint-disable-next-line react/no-array-index-key
+                                        <div key={`column-${index}`}>
+                                            {column.map(({ title, queryExamples }) => (
+                                                <QueryExamplesSection
+                                                    key={title}
+                                                    title={title}
+                                                    queryExamples={queryExamples}
+                                                    onQueryExampleClick={onQueryExampleClick}
+                                                />
+                                            ))}
+                                            {!!index && (
```

Comment:
This usage of `!!` doesn't feel intuitive to me. I'd rather change it into something a bit longer but more readable: (and also future-proof if we ever add more columns):

```suggestion
                                            {/* Put reference link at the end of the last column */}
                                            {index === exampleSyntaxColumns.length - 1 && (
```

Reply:
Yes good point! Take a look now, I added better logic with a comment for context